### PR TITLE
(7) Add invalid date test

### PR DESCRIPTION
**Pull request by aljones15**
Adds an invalid created test for the base and by extension derived proof.

TODO
- [x] Use a generator to ensure created still shows up in the proof until the implementation used for test data creation can support `proof.created`.

### CODE REVIEW

**Review by PatStLouis — COMMENTED** [review]
Looks like this PR also includes #34, it makes it difficult to see what belongs to this PR

**Comment by aljones15** [comment]
> Looks like this PR also includes #34, it makes it difficult to see what belongs to this PR

I retargeted it to point at PR 34. Sorry about that.

File: tests/suites/verify.js [hunk 1]
```diff
@@ -101,13 +101,23 @@ export function verifySuite({
             disclosed?.invalid?.proofTypeAndCryptosuite);
           await verificationFail({credential, verifier});
         });
+        it('If proofConfig.created is set and if the value is not a valid ' +
```
Comment (TallTed) [inline tests/suites/verify.js:104]:
```suggestion
        it('If proofConfig.created is set and the value is not a valid ' +
```

**Review by TallTed — DISMISSED** [review]

File: tests/suites/verify.js (hunk 1 above)
Comment (aljones15) [reply tests/suites/verify.js:104]: @TallTed we try to match the spec as exactly as possible, so I hope you don't mind, but I made a PR here: https://github.com/w3c/vc-di-bbs/pull/175/files

**Review by Wind4Greg — APPROVED** [review]

**Comment by aljones15** [comment]
@TallTed also need re-review here when ready.

File: tests/suites/verify.js [hunk 2]
```diff
@@ -135,6 +135,14 @@ export function verifySuite({
             verifier
           });
         });
+        it('If proofConfig.created is set and the value is not a valid ' +
+        '[XMLSCHEMA11-2] datetime, an INVALID_PROOF_DATETIME error MUST be ' +
+        'raised.', async function() {
```
Comment (TallTed) [inline tests/suites/verify.js:140]: I think this no longer matches wording of spec, which I think now has consistent wording across all such statements, e.g., `if {condition}, an error MUST be raised which SHOULD communicate {error_type}`.

Comment (aljones15) [reply tests/suites/verify.js:140]: you are correct the spec text has changed to: `If proofConfig.created is set and if the value is not a valid [[XMLSCHEMA11-2](https://w3c.github.io/vc-di-bbs/#bib-xmlschema11-2)] datetime, an error MUST be raised and SHOULD convey an error type of [PROOF_GENERATION_ERROR](https://www.w3.org/TR/vc-data-integrity/#PROOF_GENERATION_ERROR).`

Comment (aljones15) [reply tests/suites/verify.js]: @TallTed thanks for the catch and updated here: https://github.com/w3c/vc-di-bbs-test-suite/pull/33/commits/46e767307e12482bea5ada24d2f0d16e73bd011f

**Review by PatStLouis — APPROVED** [review]

File: tests/suites/verify.js (hunk 1 above)
Comment (TallTed) [reply tests/suites/verify.js:139]: I think this should not be marked resolved until https://github.com/w3c/vc-di-bbs/issues/178 is resolved.

Comment (aljones15) [reply tests/suites/verify.js:139]: @TallTed as this PR is part of a chain with other tests in it perhaps we should merge this as is and fix once that issue has been addressed and the group can agree on consensus on spec text? The big issue I'm running into here is that there are 4 more PRs branched from this branch which build on the features uses in this PR, and some of those PRs contain tests and features that would improve main and make the suite extensible. So it seems like we need to monitor and update test titles more.

Comment (TallTed) [reply tests/suites/verify.js:139]: OK; merge this one. Agreement between test and spec language is now being tracked in https://github.com/w3c/vc-di-bbs-test-suite/issues/48

**Comment by TallTed** [comment]
> this PR has been dangling for 2 weeks

Hardly. It's been actively worked on for 2 weeks. It might be considered to have "dangled" for 1 day (since @PatStLouis's approving review), but I think that the "dangling" time is closer to 2 hours, which was more like 1 hour when the above line was written.

**Comment by aljones15** [comment]
> > this PR has been dangling for 2 weeks
> 
> Hardly. It's been actively worked on for 2 weeks. It might be considered to have "dangled" for 1 day (since @PatStLouis's approving review), but I think that the "dangling" time is closer to 2 hours, which was more like 1 hour when the above line was written.

my apologies @TallTed and no offense meant.

**Comment by TallTed** [comment]
Accepted.

**Pull request merged**